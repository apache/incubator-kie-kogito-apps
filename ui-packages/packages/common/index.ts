export {
  default as DataTable
} from './src/components/Organisms/DataTable/DataTable';
export {
  default as KogitoSpinner
} from './src/components/Atoms/KogitoSpinner/KogitoSpinner';
export {
  default as PageToolbar
} from './src/components/Molecules/PageToolbar/PageToolbar';
export {
  default as KogitoPageLayout
} from './src/components/Templates/KogitoPageLayout/KogitoPageLayout';
export {
  default as AboutModalBox
} from './src/components/Molecules/AboutModalBox/AboutModalBox';
export {
  default as EndpointLink
} from './src/components/Atoms/EndpointLink/EndpointLink';
export {
  default as PageNotFound
} from './src/components/Molecules/PageNotFound/PageNotFound';
export {
  default as ServerUnavailable
} from './src/components/Molecules/ServerUnavailable/ServerUnavailable';
export { default as NoData } from './src/components/Molecules/NoData/NoData';
export {
  default as ServerErrors
} from './src/components/Molecules/ServerErrors/ServerErrors';
export {
  default as ProcessDescriptor
} from './src/components/Molecules/ProcessDescriptor/ProcessDescriptor';
export * from './src/components/Atoms/KogitoEmptyState/KogitoEmptyState';
export { default as LoadMore } from './src/components/Atoms/LoadMore/LoadMore';
export {
  default as DomainExplorer
} from './src/components/Organisms/DomainExplorer/DomainExplorer';
export {
  default as DomainExplorerListDomains
} from './src/components/Organisms/DomainExplorerListDomains/DomainExplorerListDomains';
export * from './src/utils/OuiaUtils';
export * from './src/graphql/types';
