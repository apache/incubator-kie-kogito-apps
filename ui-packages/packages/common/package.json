{
  "name": "@kogito-apps/common",
  "version": "1.0.0",
  "private": true,
  "main": "index.js",
  "author": "",
  "license": "Apache-2.0",
  "repository": {
    "type": "git",
    "url": "https://github.com/kiegroup/kogito-apps.git"
  },
  "scripts": {
    "precommit": "lint-staged",
    "build:prod": "yarn run lint && yarn run test:coverage",
    "test:report": "yarn test --ci --reporters=jest-junit",
    "test": "jest --runInBand --ci --reporters=default --reporters=jest-junit",
    "update-snapshot": "jest --updateSnapshot",
    "test:coverage": "yarn test --coverage",
    "storybook": "start-storybook -p 9001 -c .storybook",
    "lint": "tslint -c ./tslint.json --project . './src/**/*.ts{,x}'",
    "format": "prettier --config ../../.prettierrc --check --write './src/**/*.{tsx,ts,js}'",
    "clean": "rimraf dist",
    "codegen": "graphql-codegen"
  },
  "dependencies": {},
  "devDependencies": {
    "@storybook/addon-a11y": "5.3.19",
    "@storybook/addon-actions": "5.3.18",
    "@storybook/addon-backgrounds": "5.3.19",
    "@storybook/addon-console": "1.2.1",
    "@storybook/addon-knobs": "5.3.18",
    "@storybook/addon-links": "5.3.18",
    "@storybook/addon-options": "5.3.18",
    "@storybook/addon-storysource": "5.3.18",
    "@storybook/addon-viewport": "5.3.18",
    "@storybook/react": "5.3.18",
    "@storybook/source-loader": "5.3.18",
    "babel-loader": "8.1.0",
    "react-docgen-typescript-loader": "3.7.2",
    "storybook-react-router": "1.0.8"
  },
  "jest": {
    "preset": "ts-jest/presets/js-with-ts",
    "setupFiles": [
      "../../config/Jest-config/test-shim.js",
      "../../config/Jest-config/test-setup.js",
      "core-js"
    ],
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js"
    ],
    "coveragePathIgnorePatterns": [
      "./src/static",
      "./src/graphql"
    ],
    "snapshotSerializers": [
      "enzyme-to-json/serializer"
    ],
    "transform": {
      "^.+\\.(ts|tsx)$": "ts-jest",
      "\\.(jpg|jpeg|png|svg)$": "../../config/Jest-config/fileMocks.js"
    },
    "testMatch": [
      "**/tests/*.(ts|tsx|js)"
    ],
    "moduleNameMapper": {
      "\\.(scss|sass|css)$": "identity-obj-proxy"
    }
  },
  "lint-staged": {
    "*.{ts,tsx}": [
      "yarn run format",
      "yarn run lint"
    ]
  }
}
