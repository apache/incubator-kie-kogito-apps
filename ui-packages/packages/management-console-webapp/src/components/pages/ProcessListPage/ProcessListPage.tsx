/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React, { useEffect } from 'react';
import { Card } from '@patternfly/react-core/dist/js/components/Card';
import { PageSection } from '@patternfly/react-core/dist/js/components/Page';
import {
  OUIAProps,
  ouiaPageTypeAndObjectId,
  componentOuiaProps
} from '@kogito-apps/ouia-tools/dist/utils/OuiaUtils';
import { RouteComponentProps } from 'react-router-dom';
import { PageSectionHeader } from '@kogito-apps/consoles-common/dist/components/layout/PageSectionHeader';
import ProcessListContainer from '../../containers/ProcessListContainer/ProcessListContainer';
import { StaticContext } from 'react-router';
import * as H from 'history';
import '../../styles.css';
import { ProcessListState } from '@kogito-apps/management-console-shared/dist/types';

interface MatchProps {
  instanceID: string;
}

const ProcessListPage: React.FC<
  RouteComponentProps<MatchProps, StaticContext, H.LocationState> & OUIAProps
> = (props) => {
  useEffect(() => {
    return ouiaPageTypeAndObjectId('process-instances');
  });
  const initialState: ProcessListState =
    props.location && (props.location.state as ProcessListState);

  return (
    <React.Fragment>
      <PageSectionHeader
        titleText="Process Instances"
        breadcrumbText={['Home', 'Processes']}
        breadcrumbPath={['/']}
        ouiaId={props.ouiaId}
      />
      <PageSection
        {...componentOuiaProps(
          props.ouiaId,
          'page-section-content',
          props.ouiaSafe
        )}
      >
        <Card className="kogito-management-console__card-size">
          <ProcessListContainer initialState={initialState} />
        </Card>
      </PageSection>
    </React.Fragment>
  );
};

export default ProcessListPage;
