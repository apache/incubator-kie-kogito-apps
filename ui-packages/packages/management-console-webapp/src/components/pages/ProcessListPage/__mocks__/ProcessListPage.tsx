import React from 'react';

const MockedProcessListPage: React.FC = () => {
  return <></>;
};

export default MockedProcessListPage;
