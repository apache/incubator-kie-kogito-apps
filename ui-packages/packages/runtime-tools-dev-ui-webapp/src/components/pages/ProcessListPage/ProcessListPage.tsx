/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { useEffect } from 'react';
import { Card, PageSection } from '@patternfly/react-core';
import {
  OUIAProps,
  ouiaPageTypeAndObjectId
} from '@kogito-apps/components-common';
import { RouteComponentProps } from 'react-router-dom';
import * as H from 'history';
import { PageSectionHeader } from '@kogito-apps/consoles-common';
import ProcessListContainer from '../../containers/ProcessListContainer/ProcessListContainer';
import '../../styles.css';
import { ProcessListState } from '@kogito-apps/management-console-shared';

const ProcessListPage: React.FC<RouteComponentProps<H.LocationState> &
  OUIAProps> = props => {
  useEffect(() => {
    return ouiaPageTypeAndObjectId('jobs-management');
  });

  const initialState: ProcessListState =
    props.location && (props.location.state as ProcessListState);

  return (
    <React.Fragment>
      <PageSectionHeader
        titleText="Process Instances"
        breadcrumbText={['Home', 'process-instances']}
        breadcrumbPath={['/']}
      />
      <PageSection>
        <Card className="kogito-consoles__card-size">
          <ProcessListContainer initialState={initialState} />
        </Card>
      </PageSection>
    </React.Fragment>
  );
};

export default ProcessListPage;
