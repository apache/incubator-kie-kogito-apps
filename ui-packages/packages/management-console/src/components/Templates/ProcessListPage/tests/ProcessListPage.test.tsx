import React from 'react';
import ProcessListPage from '../ProcessListPage';
import { GraphQL, getWrapperAsync } from '@kogito-apps/common';
import { MockedProvider } from '@apollo/react-testing';
import { BrowserRouter } from 'react-router-dom';
import { Button, EmptyStateBody, EmptyState } from '@patternfly/react-core';
import { act } from 'react-dom/test-utils';
jest.mock('../../../Organisms/ProcessListTable/ProcessListTable');
jest.mock('../../../Atoms/ProcessListBulkInstances/ProcessListBulkInstances');
jest.mock('../../../Atoms/ProcessListModal/ProcessListModal');
const MockedComponent = (): React.ReactElement => {
  return <></>;
};
jest.mock('@kogito-apps/common', () => ({
  ...jest.requireActual('@kogito-apps/common'),
  LoadMore: () => {
    return <MockedComponent />;
  }
}));
jest.mock('@patternfly/react-icons', () => ({
  ...jest.requireActual('@patternfly/react-icons'),
  ExclamationTriangleIcon: () => {
    return <MockedComponent />;
  },
  ExclamationCircleIcon: () => {
    return <MockedComponent />;
  }
}));

const routeComponentPropsMock1 = {
  history: {} as any,
  location: {
    pathname: '/ProcessInstances',
    state: {
      filters: {
        status: [GraphQL.ProcessInstanceState.Active],
        businessKey: []
      }
    }
  } as any,
  match: {
    params: {}
  } as any
};

const routeComponentPropsMock2 = {
  history: {} as any,
  location: {
    pathname: '/ProcessInstances',
    state: {
      filters: {
        status: [GraphQL.ProcessInstanceState.Active],
        businessKey: ['MQQ640']
      }
    }
  } as any,
  match: {
    params: {}
  } as any
};

const routeComponentPropsMock3 = {
  history: {} as any,
  location: {
    pathname: '/ProcessInstances',
    state: {
      filters: {
        status: [],
        businessKey: []
      }
    }
  } as any,
  match: {
    params: {}
  } as any
};

const mocks1 = [
  {
    request: {
      query: GraphQL.GetProcessInstancesDocument,
      variables: {
        state: [GraphQL.ProcessInstanceState.Active],
        offset: 0,
        limit: 10
      }
    },
    result: {
      data: {
        ProcessInstances: [
          {
            id: '201a8a42-043e-375a-9f52-57c804b24db4',
            processId: 'travels',
            processName: 'travels',
            businessKey: null,
            rootProcessInstanceId: null,
            parentProcessInstanceId: null,
            parentProcessInstance: null,
            roles: [],
            variables:
              '{"flight":{"flightNumber":"MX555","seat":null,"gate":null,"departure":"2020-05-08T03:30:00.000+05:30","arrival":"2020-05-09T03:30:00.000+05:30"},"hotel":{"name":"Perfect hotel","address":{"street":"street","city":"Bengaluru","zipCode":"12345","country":"India"},"phone":"09876543","bookingNumber":"XX-012345","room":null},"trip":{"city":"Bengaluru","country":"India","begin":"2020-05-08T03:30:00.000+05:30","end":"2020-05-09T03:30:00.000+05:30","visaRequired":false},"traveller":{"firstName":"Ajay","lastName":"Jaganathan","email":"ajaganat@redhat.com","nationality":"Polish","address":{"street":"Bangalore","city":"Bangalore","zipCode":"560093","country":"Poland"}}}',
            state: GraphQL.ProcessInstanceState.Active,
            start: '2020-05-07T06:50:18.274Z',
            lastUpdate: '2020-05-07T06:50:18.502Z',
            end: null,
            addons: [
              'process-management',
              'infinispan-persistence',
              'prometheus-monitoring'
            ],
            endpoint: 'http://localhost:8080/travels',
            serviceUrl: 'http://localhost:8080',
            error: null,
            childProcessInstances: [
              {
                id: 'bfde98ed-0cdd-4700-ae87-377f7ec430cd',
                processName: 'HotelBooking',
                businessKey: null
              },
              {
                id: 'e607b2a9-0aca-4788-9623-dd2e156ce9c4',
                processName: 'FlightBooking',
                businessKey: null
              }
            ],
            nodes: [
              {
                id: '39d5fe7c-4e37-44ce-8d25-05a4a29ec6ea',
                nodeId: '17',
                name: 'Book Hotel',
                enter: '2020-05-07T06:50:18.429Z',
                exit: '2020-05-07T06:50:18.439Z',
                type: 'SubProcessNode',
                definitionId: '_1A708F87-11C0-42A0-A464-0B7E259C426F'
              },
              {
                id: '1d3d7ebe-79ec-4848-b1d6-d7c5a371b4dd',
                nodeId: '8',
                name: 'Confirm travel',
                enter: '2020-05-07T06:50:18.443Z',
                exit: null,
                type: 'HumanTaskNode',
                definitionId: '_2140F05A-364F-40B3-BB7B-B12927065DF8'
              },
              {
                id: '4fc774f5-c429-4b97-8981-bd60d1f59954',
                nodeId: '7',
                name: 'Join',
                enter: '2020-05-07T06:50:18.442Z',
                exit: '2020-05-07T06:50:18.443Z',
                type: 'Join',
                definitionId: '_5D0733B5-53FE-40E9-9900-4CC13419C67A'
              },
              {
                id: '60309b28-1dbe-49c3-b0d8-8cefb86be8b4',
                nodeId: '10',
                name: 'Book Flight',
                enter: '2020-05-07T06:50:18.439Z',
                exit: '2020-05-07T06:50:18.442Z',
                type: 'SubProcessNode',
                definitionId: '_F543B3F0-AB44-4A5B-BF17-8D9DEB505815'
              },
              {
                id: 'efa5a6c0-9470-4405-8609-620bdbfbb6d9',
                nodeId: '2',
                name: 'Book',
                enter: '2020-05-07T06:50:18.428Z',
                exit: '2020-05-07T06:50:18.439Z',
                type: 'Split',
                definitionId: '_175DC79D-C2F1-4B28-BE2D-B583DFABF70D'
              },
              {
                id: 'acc87f1f-e436-4364-b4d3-9069ddb359c9',
                nodeId: '14',
                name: 'Join',
                enter: '2020-05-07T06:50:18.428Z',
                exit: '2020-05-07T06:50:18.428Z',
                type: 'Join',
                definitionId: '_B34ADDEE-DEA5-47C5-A913-F8B85ED5641F'
              },
              {
                id: '6c20729a-3e2b-44c5-af5c-8554f21a7ebf',
                nodeId: '15',
                name: 'is visa required',
                enter: '2020-05-07T06:50:18.427Z',
                exit: '2020-05-07T06:50:18.428Z',
                type: 'Split',
                definitionId: '_5EA95D17-59A6-4567-92DF-74D36CE7F35A'
              },
              {
                id: '3fb92a0a-e88b-45a4-a48b-65f11a679b96',
                nodeId: '5',
                name: 'Visa check',
                enter: '2020-05-07T06:50:18.278Z',
                exit: '2020-05-07T06:50:18.427Z',
                type: 'RuleSetNode',
                definitionId: '_54ABE1ED-61BE-45F9-812C-795A5D4ED35E'
              },
              {
                id: '2860e474-dffc-498c-8731-e6b6f0b5d4d4',
                nodeId: '16',
                name: 'StartProcess',
                enter: '2020-05-07T06:50:18.276Z',
                exit: '2020-05-07T06:50:18.278Z',
                type: 'StartNode',
                definitionId: '_1B11BEC9-402A-4E73-959A-296BD334CAB0'
              }
            ]
          }
        ]
      }
    }
  }
];

const mocks2 = [
  {
    request: {
      query: GraphQL.GetProcessInstancesWithBusinessKeyDocument,
      variables: {
        state: [GraphQL.ProcessInstanceState.Active],
        offset: 0,
        limit: 10,
        businessKeys: [{ businessKey: { like: 'MQQ640' } }]
      }
    },
    result: {
      data: {
        ProcessInstances: [
          {
            id: '201a8a42-043e-375a-9f52-57c804b24db4',
            processId: 'travels',
            processName: 'travels',
            businessKey: 'MQQ640',
            rootProcessInstanceId: null,
            parentProcessInstanceId: null,
            parentProcessInstance: null,
            roles: [],
            variables:
              '{"flight":{"flightNumber":"MX555","seat":null,"gate":null,"departure":"2020-05-08T03:30:00.000+05:30","arrival":"2020-05-09T03:30:00.000+05:30"},"hotel":{"name":"Perfect hotel","address":{"street":"street","city":"Bengaluru","zipCode":"12345","country":"India"},"phone":"09876543","bookingNumber":"XX-012345","room":null},"trip":{"city":"Bengaluru","country":"India","begin":"2020-05-08T03:30:00.000+05:30","end":"2020-05-09T03:30:00.000+05:30","visaRequired":false},"traveller":{"firstName":"Ajay","lastName":"Jaganathan","email":"ajaganat@redhat.com","nationality":"Polish","address":{"street":"Bangalore","city":"Bangalore","zipCode":"560093","country":"Poland"}}}',
            state: 'ACTIVE',
            start: '2020-05-07T06:50:18.274Z',
            lastUpdate: '2020-05-07T06:50:18.502Z',
            end: null,
            addons: [
              'process-management',
              'infinispan-persistence',
              'prometheus-monitoring'
            ],
            endpoint: 'http://localhost:8080/travels',
            serviceUrl: 'http://localhost:8080',
            error: null,
            childProcessInstances: [
              {
                id: 'bfde98ed-0cdd-4700-ae87-377f7ec430cd',
                processName: 'HotelBooking',
                businessKey: null
              },
              {
                id: 'e607b2a9-0aca-4788-9623-dd2e156ce9c4',
                processName: 'FlightBooking',
                businessKey: null
              }
            ],
            nodes: [
              {
                id: '39d5fe7c-4e37-44ce-8d25-05a4a29ec6ea',
                nodeId: '17',
                name: 'Book Hotel',
                enter: '2020-05-07T06:50:18.429Z',
                exit: '2020-05-07T06:50:18.439Z',
                type: 'SubProcessNode',
                definitionId: '_1A708F87-11C0-42A0-A464-0B7E259C426F'
              },
              {
                id: '1d3d7ebe-79ec-4848-b1d6-d7c5a371b4dd',
                nodeId: '8',
                name: 'Confirm travel',
                enter: '2020-05-07T06:50:18.443Z',
                exit: null,
                type: 'HumanTaskNode',
                definitionId: '_2140F05A-364F-40B3-BB7B-B12927065DF8'
              },
              {
                id: '4fc774f5-c429-4b97-8981-bd60d1f59954',
                nodeId: '7',
                name: 'Join',
                enter: '2020-05-07T06:50:18.442Z',
                exit: '2020-05-07T06:50:18.443Z',
                type: 'Join',
                definitionId: '_5D0733B5-53FE-40E9-9900-4CC13419C67A'
              },
              {
                id: '60309b28-1dbe-49c3-b0d8-8cefb86be8b4',
                nodeId: '10',
                name: 'Book Flight',
                enter: '2020-05-07T06:50:18.439Z',
                exit: '2020-05-07T06:50:18.442Z',
                type: 'SubProcessNode',
                definitionId: '_F543B3F0-AB44-4A5B-BF17-8D9DEB505815'
              },
              {
                id: 'efa5a6c0-9470-4405-8609-620bdbfbb6d9',
                nodeId: '2',
                name: 'Book',
                enter: '2020-05-07T06:50:18.428Z',
                exit: '2020-05-07T06:50:18.439Z',
                type: 'Split',
                definitionId: '_175DC79D-C2F1-4B28-BE2D-B583DFABF70D'
              },
              {
                id: 'acc87f1f-e436-4364-b4d3-9069ddb359c9',
                nodeId: '14',
                name: 'Join',
                enter: '2020-05-07T06:50:18.428Z',
                exit: '2020-05-07T06:50:18.428Z',
                type: 'Join',
                definitionId: '_B34ADDEE-DEA5-47C5-A913-F8B85ED5641F'
              },
              {
                id: '6c20729a-3e2b-44c5-af5c-8554f21a7ebf',
                nodeId: '15',
                name: 'is visa required',
                enter: '2020-05-07T06:50:18.427Z',
                exit: '2020-05-07T06:50:18.428Z',
                type: 'Split',
                definitionId: '_5EA95D17-59A6-4567-92DF-74D36CE7F35A'
              },
              {
                id: '3fb92a0a-e88b-45a4-a48b-65f11a679b96',
                nodeId: '5',
                name: 'Visa check',
                enter: '2020-05-07T06:50:18.278Z',
                exit: '2020-05-07T06:50:18.427Z',
                type: 'RuleSetNode',
                definitionId: '_54ABE1ED-61BE-45F9-812C-795A5D4ED35E'
              },
              {
                id: '2860e474-dffc-498c-8731-e6b6f0b5d4d4',
                nodeId: '16',
                name: 'StartProcess',
                enter: '2020-05-07T06:50:18.276Z',
                exit: '2020-05-07T06:50:18.278Z',
                type: 'StartNode',
                definitionId: '_1B11BEC9-402A-4E73-959A-296BD334CAB0'
              }
            ]
          }
        ]
      }
    }
  }
];

const mocks3 = [
  {
    request: {
      query: GraphQL.GetProcessInstancesDocument,
      variables: {
        state: [GraphQL.ProcessInstanceState.Active],
        offset: 0,
        limit: 10
      }
    },
    error: new Error('something went wrong')
  }
];

const mocks5 = [
  {
    request: {
      query: GraphQL.GetProcessInstancesDocument,
      variables: {
        state: [],
        offset: 0,
        limit: 10
      }
    },
    result: {
      data: {
        ProcessInstances: []
      }
    }
  }
];

const mocks4 = [
  {
    request: {
      query: GraphQL.GetProcessInstancesWithBusinessKeyDocument,
      variables: {
        state: [GraphQL.ProcessInstanceState.Active],
        offset: 0,
        limit: 10,
        businessKeys: [{ businessKey: { like: 'MQQ640' } }]
      }
    },
    error: new Error('something went wrong')
  }
];
describe('ProcessListPage component tests', () => {
  /* tslint:disable */
  it('on FilterClick tests- without businesskey', async () => {
    let wrapper = await getWrapperAsync(
      <BrowserRouter>
        <MockedProvider mocks={mocks1} addTypename={false}>
          <ProcessListPage {...routeComponentPropsMock1} />
        </MockedProvider>
      </BrowserRouter>,
      'ProcessListPage'
    );
    await act(async () => {
      wrapper
        .find('#apply-filter-button')
        .find(Button)
        .find('button')
        .simulate('click');
    });
    wrapper = wrapper.update();
    expect(
      wrapper.find('MockedProcessListTable').props()['initData'][
        'ProcessInstances'
      ][0]['id']
    ).toEqual(mocks1[0].result.data.ProcessInstances[0].id);
  });

  it('on FilterClick tests- with businesskey', async () => {
    let wrapper = await getWrapperAsync(
      <BrowserRouter>
        <MockedProvider mocks={mocks2} addTypename={false}>
          <ProcessListPage {...routeComponentPropsMock2} />
        </MockedProvider>
      </BrowserRouter>,
      'ProcessListPage'
    );
    await act(async () => {
      wrapper
        .find('#apply-filter-button')
        .find(Button)
        .find('button')
        .simulate('click');
    });
    wrapper = wrapper.update();
    expect(
      wrapper.find('MockedProcessListTable').props()['initData'][
        'ProcessInstances'
      ][0]['businessKey']
    ).toEqual(mocks2[0].result.data.ProcessInstances[0].businessKey);
  });

  it('error in query - without businesskey', async () => {
    let wrapper = await getWrapperAsync(
      <BrowserRouter>
        <MockedProvider mocks={mocks3} addTypename={false}>
          <ProcessListPage {...routeComponentPropsMock1} />
        </MockedProvider>
      </BrowserRouter>,
      'ProcessListPage'
    );
    await act(async () => {
      wrapper
        .find('#apply-filter-button')
        .find(Button)
        .find('button')
        .simulate('click');
    });
    wrapper = wrapper.update();
    wrapper = wrapper.find(EmptyState);
    expect(
      wrapper
        .find(EmptyStateBody)
        .children()
        .html()
        .includes('An error occurred while accessing data.')
    ).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  it('error in query - with businesskey', async () => {
    let wrapper = await getWrapperAsync(
      <BrowserRouter>
        <MockedProvider mocks={mocks4} addTypename={false}>
          <ProcessListPage {...routeComponentPropsMock2} />
        </MockedProvider>
      </BrowserRouter>,
      'ProcessListPage'
    );
    await act(async () => {
      wrapper
        .find('#apply-filter-button')
        .find(Button)
        .find('button')
        .simulate('click');
    });
    wrapper = wrapper.update();
    wrapper = wrapper.find(EmptyState);
    expect(
      wrapper
        .find(EmptyStateBody)
        .children()
        .html()
        .includes('An error occurred while accessing data.')
    ).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  it('Snapshot tests for no status selected', async () => {
    let wrapper = await getWrapperAsync(
      <BrowserRouter>
        <MockedProvider mocks={mocks5} addTypename={false}>
          <ProcessListPage {...routeComponentPropsMock3} />
        </MockedProvider>
      </BrowserRouter>,
      'ProcessListPage'
    );
    wrapper = wrapper.find(EmptyState);
    expect(wrapper).toMatchSnapshot();
  });

  it('reset click in no status found test', async () => {
    let wrapper = await getWrapperAsync(
      <BrowserRouter>
        <MockedProvider mocks={mocks5} addTypename={false}>
          <ProcessListPage {...routeComponentPropsMock3} />
        </MockedProvider>
      </BrowserRouter>,
      'ProcessListPage'
    );
    await act(async () => {
      wrapper
        .find(EmptyState)
        .find(Button)
        .find('button')
        .simulate('click');
    });
  });
});
