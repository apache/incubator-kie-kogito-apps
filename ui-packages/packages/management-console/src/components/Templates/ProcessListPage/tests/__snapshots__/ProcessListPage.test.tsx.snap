// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProcessListPage component tests Snapshot tests for no status selected 1`] = `
<EmptyState
  variant="full"
>
  <div
    className="pf-c-empty-state"
  >
    <EmptyStateIcon
      color="var(--pf-global--warning-color--100)"
      icon={[Function]}
      size="sm"
    >
      <ExclamationTriangleIcon
        aria-hidden="true"
        className="pf-c-empty-state__icon"
        color="var(--pf-global--warning-color--100)"
        size="sm"
      >
        <MockedComponent />
      </ExclamationTriangleIcon>
    </EmptyStateIcon>
    <Title
      headingLevel="h5"
      size="lg"
    >
      <h5
        className="pf-c-title pf-m-lg"
      >
        No status is selected
      </h5>
    </Title>
    <EmptyStateBody>
      <div
        className="pf-c-empty-state__body"
      >
        Try selecting at least one status to see results
      </div>
    </EmptyStateBody>
    <Component
      onClick={[Function]}
      variant="link"
    >
      <ComponentWithOuia
        component={[Function]}
        componentProps={
          Object {
            "children": "Reset to default",
            "onClick": [Function],
            "variant": "link",
          }
        }
        consumerContext={
          Object {
            "isOuia": false,
            "ouiaId": null,
          }
        }
      >
        <Button
          onClick={[Function]}
          ouiaContext={
            Object {
              "isOuia": false,
              "ouiaId": null,
            }
          }
          variant="link"
        >
          <button
            aria-disabled={null}
            aria-label={null}
            className="pf-c-button pf-m-link"
            disabled={false}
            onClick={[Function]}
            tabIndex={null}
            type="button"
          >
            <span
              className="pf-c-button__text"
            >
              Reset to default
            </span>
          </button>
        </Button>
      </ComponentWithOuia>
    </Component>
  </div>
</EmptyState>
`;

exports[`ProcessListPage component tests error in query - with businesskey 1`] = `
<EmptyState
  variant="full"
>
  <div
    className="pf-c-empty-state"
  >
    <EmptyStateIcon
      color="var(--pf-global--danger-color--100)"
      icon={[Function]}
      size="md"
    >
      <ExclamationCircleIcon
        aria-hidden="true"
        className="pf-c-empty-state__icon"
        color="var(--pf-global--danger-color--100)"
        size="md"
      >
        <MockedComponent />
      </ExclamationCircleIcon>
    </EmptyStateIcon>
    <Title
      headingLevel="h1"
      size="4xl"
    >
      <h1
        className="pf-c-title pf-m-4xl"
      >
        Error fetching data
      </h1>
    </Title>
    <EmptyStateBody>
      <div
        className="pf-c-empty-state__body"
      >
        An error occurred while accessing data.
         
        <Component
          id="display-error"
          isInline={true}
          onClick={[Function]}
          variant="link"
        >
          <ComponentWithOuia
            component={[Function]}
            componentProps={
              Object {
                "children": "See more details",
                "id": "display-error",
                "isInline": true,
                "onClick": [Function],
                "variant": "link",
              }
            }
            consumerContext={
              Object {
                "isOuia": false,
                "ouiaId": null,
              }
            }
          >
            <Button
              id="display-error"
              isInline={true}
              onClick={[Function]}
              ouiaContext={
                Object {
                  "isOuia": false,
                  "ouiaId": null,
                }
              }
              variant="link"
            >
              <button
                aria-disabled={null}
                aria-label={null}
                className="pf-c-button pf-m-link pf-m-inline"
                disabled={false}
                id="display-error"
                onClick={[Function]}
                tabIndex={null}
                type="button"
              >
                <span
                  className="pf-c-button__text"
                >
                  See more details
                </span>
              </button>
            </Button>
          </ComponentWithOuia>
        </Component>
      </div>
    </EmptyStateBody>
    <Component
      id="goback-button"
      onClick={[Function]}
      variant="primary"
    >
      <ComponentWithOuia
        component={[Function]}
        componentProps={
          Object {
            "children": "Go back",
            "id": "goback-button",
            "onClick": [Function],
            "variant": "primary",
          }
        }
        consumerContext={
          Object {
            "isOuia": false,
            "ouiaId": null,
          }
        }
      >
        <Button
          id="goback-button"
          onClick={[Function]}
          ouiaContext={
            Object {
              "isOuia": false,
              "ouiaId": null,
            }
          }
          variant="primary"
        >
          <button
            aria-disabled={null}
            aria-label={null}
            className="pf-c-button pf-m-primary"
            disabled={false}
            id="goback-button"
            onClick={[Function]}
            tabIndex={null}
            type="button"
          >
            Go back
          </button>
        </Button>
      </ComponentWithOuia>
    </Component>
  </div>
</EmptyState>
`;

exports[`ProcessListPage component tests error in query - without businesskey 1`] = `
<EmptyState
  variant="full"
>
  <div
    className="pf-c-empty-state"
  >
    <EmptyStateIcon
      color="var(--pf-global--danger-color--100)"
      icon={[Function]}
      size="md"
    >
      <ExclamationCircleIcon
        aria-hidden="true"
        className="pf-c-empty-state__icon"
        color="var(--pf-global--danger-color--100)"
        size="md"
      >
        <MockedComponent />
      </ExclamationCircleIcon>
    </EmptyStateIcon>
    <Title
      headingLevel="h1"
      size="4xl"
    >
      <h1
        className="pf-c-title pf-m-4xl"
      >
        Error fetching data
      </h1>
    </Title>
    <EmptyStateBody>
      <div
        className="pf-c-empty-state__body"
      >
        An error occurred while accessing data.
         
        <Component
          id="display-error"
          isInline={true}
          onClick={[Function]}
          variant="link"
        >
          <ComponentWithOuia
            component={[Function]}
            componentProps={
              Object {
                "children": "See more details",
                "id": "display-error",
                "isInline": true,
                "onClick": [Function],
                "variant": "link",
              }
            }
            consumerContext={
              Object {
                "isOuia": false,
                "ouiaId": null,
              }
            }
          >
            <Button
              id="display-error"
              isInline={true}
              onClick={[Function]}
              ouiaContext={
                Object {
                  "isOuia": false,
                  "ouiaId": null,
                }
              }
              variant="link"
            >
              <button
                aria-disabled={null}
                aria-label={null}
                className="pf-c-button pf-m-link pf-m-inline"
                disabled={false}
                id="display-error"
                onClick={[Function]}
                tabIndex={null}
                type="button"
              >
                <span
                  className="pf-c-button__text"
                >
                  See more details
                </span>
              </button>
            </Button>
          </ComponentWithOuia>
        </Component>
      </div>
    </EmptyStateBody>
    <Component
      id="goback-button"
      onClick={[Function]}
      variant="primary"
    >
      <ComponentWithOuia
        component={[Function]}
        componentProps={
          Object {
            "children": "Go back",
            "id": "goback-button",
            "onClick": [Function],
            "variant": "primary",
          }
        }
        consumerContext={
          Object {
            "isOuia": false,
            "ouiaId": null,
          }
        }
      >
        <Button
          id="goback-button"
          onClick={[Function]}
          ouiaContext={
            Object {
              "isOuia": false,
              "ouiaId": null,
            }
          }
          variant="primary"
        >
          <button
            aria-disabled={null}
            aria-label={null}
            className="pf-c-button pf-m-primary"
            disabled={false}
            id="goback-button"
            onClick={[Function]}
            tabIndex={null}
            type="button"
          >
            Go back
          </button>
        </Button>
      </ComponentWithOuia>
    </Component>
  </div>
</EmptyState>
`;
